routing/osrm_path_segment_factory.cpp

```diff
@@ -143,40 +144,77 @@ void OsrmPathSegmentFactory(RoutingMapping & mapping, Index const & index, RawPa
     return distance(buffer.begin(), it);
   };
 
-  // Calculate estimated time for a start and a end node cases.
-  if (isStartNode && isEndNode)
+  size_t const startIndex = isStartNode ? findIntersectingSeg(startGraphNode.segment) : 0;
+  size_t const endIndex = isEndNode ? findIntersectingSeg(endGraphNode.segment) + 1 : buffer.size();
+  // @TODO(bykoianko) When start and end points belong to the same NodeID it's possible
+  // to get endIndex <= startIndex. Production version returns route building error.
+  // Debug version crashes with an assert. The test to show it USALosAnglesAriaTwentyninePalmsHighwayTimeTest.
+  LoadPathGeometry(buffer, startIndex, endIndex, index, mapping, startGraphNode, endGraphNode, isStartNode,
+                   isEndNode, loadedPathSegment);
+
+  LoadedPathSegment fullPathSegment;
+  LoadPathGeometry(buffer, 0, buffer.size(), index, mapping, FeatureGraphNode(), FeatureGraphNode(),
```

Comment:
Here I need to get all points of NodeID (osrm edge) based on "buffer_vector<TSeg, 8> buffer". To do that I need to use LoadPathGeometry(...) method. This method has more or less strange interface. My call is
```
  LoadPathGeometry(buffer, 0, buffer.size(), index, mapping, FeatureGraphNode(), FeatureGraphNode(), false /* isStartNode */, false /* isEndNode */, fullPathSegment);
```
It means to fill |fullPathSegment| with all points of |buffer| taking into account that it's neither a start node nor an end node (isStartNode == false and isEndNode == false).